express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('./node_modules/express');
+const app = express();
+
+const grades = [];
+let nextID = 1;
+
+const parser = express.json();
+
+app.use(parser);
+
+app.get('/api/grades', (req, res) => {
+  res.json(grades);
+});
+
+app.post('/api/grades', parser, (req, res) => {
+  const newGrade = req.body;
+  newGrade.id = nextID;
+  grades.push(newGrade);
+  nextID++;
+  res.json(newGrade);
+  res.status(201);
```

Comment:
Be sure to set the status code before sending the body. The response message is written line-by-line from the top, so as soon as you do anything that sends or starts sending the body, you can't modify the status code or headers anymore.